SimPEG/utils/mat_utils.py

```diff
@@ -100,6 +100,83 @@ def uniqueRows(M):
     return unqM, unqInd, invInd
 
 
+def eigenvalue_by_power_iteration(combo_objfct, model, n_pw_iter=4, fields_list=None, seed=None):
+    """
+    Estimate the highest eigenvalue of any objective function term or combination thereof
+    (data_misfit, regularization or ComboObjectiveFunction) for a given model.
+    The highest eigenvalue is estimated by power iterations and Rayleigh quotient.
+
+    Parameters
+    ----------
+
+    :param SimPEG.BaseObjectiveFunction combo_objfct: objective function term of which to estimate the highest eigenvalue
+    :param numpy.ndarray model: current geophysical model to estimate the objective function derivatives at
+    :param int n_pw_iter: number of power iterations to estimate the highest eigenvalue
+    :param list fiels_list: (optional) list of fields for each data misfit term in combo_objfct. If none given,
+                            they will be evaluated within the function. If combo_objfct mixs data misfit and regularization
+                            terms, the list should contains SimPEG.fields for the data misfit terms and None for the
+                            regularization term.
+    :param int seed: Random seed for the initial random guess of eigenvector.
+
+    Return
+    ------
+
+    :return float eigenvalue: estimated value of the highest eigenvalye
+
+    """
+
+
+    if seed is not None:
+        np.random.seed(seed)
+    
+    # Initial guess for eigen-vector
+    x0 = np.random.rand(*model.shape)
+    x0 = x0 / np.linalg.norm(x0)
+ 
+    # transform to ComboObjectiveFunction if required
+    if getattr(combo_objfct, "objfcts", None) is None:
+        combo = 1. * combo_objfct
```

Comment:
Done. I checked and it caused no issue, so I implemented your recommendation.

Reply:
Yup, I'm just trying to reduce lines of code!

It can be kinda odd sometimes, especially when you're often dealing with numpy arrays.
Generally if you assigning to a variable using an `=` it will just reference whatever object is being assigned to it. So what happens here is that `combo_objfct` now points to the new object created by doing `1. * combo_objfct`. 

Assigning to slices or items of containers (np.ndarrays, dictionaries, lists, etc.) WILL modify the original item because the assignment maps to a specific internal function `x[3] = 2` actually ends up calling `x.__setitem__(2)`.